app/models/tag_rule/discount_order.rb

```diff
@@ -5,7 +5,7 @@ class TagRule::DiscountOrder < TagRule
 
   # Warning: this should only EVER be called via TagRule#apply
   def apply!
-    create_adjustment(I18n.t("discount"), subject, subject)
+    create_adjustment(I18n.t("discount"), subject)
```

Comment:
This is flagged as not being covered by tests, bt I'm 99% sure this is dead code. I'll take a closer look :ok_hand:

Reply:
This involves some serious archeology, but as far as I can tell, `TagRule::Discount` was a partially implemented (but not finished) feature from 2015, but it's been disabled and unused since 2016. It's not possible to create tag rules that actually use this class.

All instances older than 5 years have zero `TagRule::Discount` objects in the database:
```
irb(main):001:0> TagRule.where(type: 'TagRule::DiscountOrder').count
=> 0
```

All the related specs for this class have been set to `pending` for over 5 years. I think we can just delete this for now, it's not in use and potentially doesn't work...?

Reply:
Here's the line that lists classes that can actually be used, for reference:

https://github.com/openfoodfoundation/openfoodnetwork/blob/76def3a74feb71995b8102da1e07522c57ae1ba3/app/assets/javascripts/admin/tag_rules/directives/new_rule_dialog.js.coffee#L10-L16

Notes and discussions in related PRs say it doesn't work properly and has issues with taxes.

Reply:
I've made a PR to delete it here: https://github.com/openfoodfoundation/openfoodnetwork/pull/7351